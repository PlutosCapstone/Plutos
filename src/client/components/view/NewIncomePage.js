import React, { useState } from "react";
import { View, Text, StyleSheet, Pressable, TextInput } from "react-native";
import { RadioButton } from "react-native-paper";
import BackArrow from "../../assets/icons/BackArrow";

import { Dropdown } from "react-native-element-dropdown";

const dropdownData = [
  { label: "Daily", value: 1 },
  { label: "Weekly", value: 2 },
  { label: "Bi-Weekly", value: 3 },
  { label: "Monthly", value: 4 },
  { label: "Quarterly", value: 5 },
  { label: "Bi-annually", value: 6 },
  { label: "Annually", value: 7 },
];

const NewIncomePage = ({ navigation }) => {
  const [title, setTitle] = useState("");
  const [amount, setAmount] = useState("");
  const [checked, setChecked] = React.useState("once");
  const [dropdownValue, setDropdownValue] = useState(1);

  const dropdownChangeHandler = (item) => {
    setDropdownValue(item.value);
  };

  const returnHandler = () => {
    navigation.goBack();
  };

  const addIncomeHandler = () => {
    // Handle some logic to update state
    returnHandler();
  };

  return (
    <View style={styles.background}>
      <View style={styles.header}>
        <Pressable style={styles.backArrow} onPress={returnHandler}>
          <BackArrow size={35} />
        </Pressable>
        <Text style={styles.headerTitle}>New Income</Text>
      </View>
      <View style={styles.fieldBox}>
        <TextInput
          style={styles.textInput}
          value={title}
          onChangeText={setTitle}
          placeholder="Title: Add a Title"
        />
        <TextInput
          style={styles.textInput}
          onChangeText={setAmount}
          value={amount}
          placeholder="Amount ($):"
          keyboardType="numeric"
        />
        <View style={styles.radioButtonBox}>
          <View style={styles.radioButton}>
            <RadioButton
              value="once"
              status={checked === "once" ? "checked" : "unchecked"}
              onPress={() => setChecked("once")}
            />
            <Text onPress={() => setChecked("once")}>Once</Text>
          </View>
          <View style={styles.radioButton}>
            <RadioButton
              value="periodically"
              status={checked === "periodically" ? "checked" : "unchecked"}
              onPress={() => setChecked("periodically")}
            />
            <Text onPress={() => setChecked("periodically")}>Periodically</Text>
          </View>
        </View>
        {checked == "periodically" && (
          <View style={styles.dropdownBox}>
            <Text style={styles.dropdownText}>How often?:</Text>
            <Dropdown
              data={dropdownData}
              labelField="label"
              valueField="value"
              style={styles.dropdown}
              label="How often?"
              search
              searchPlaceholder="Search..."
              value={dropdownValue}
              onChange={(item) => dropdownChangeHandler(item)}
            />
          </View>
        )}

        <Pressable style={styles.submit} onPress={addIncomeHandler}>
          <Text style={styles.submitText}>Add Income</Text>
        </Pressable>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  background: {
    height: "100%",
    backgroundColor: "white",
    padding: 24,
    rowGap: 20,
  },
  header: {
    display: "flex",
    flexDirection: "row",
    position: "relative",
    justifyContent: "center",
    alignItems: "center",
  },
  backArrow: {
    position: "absolute",
    left: 0,
  },
  headerTitle: {
    fontWeight: "600",
    fontSize: 32,
  },
  textInput: {
    height: 50,
    borderBlockColor: "black",
    borderWidth: 1,
    borderRadius: 16,
    padding: 12,
    fontSize: 16,
  },
  fieldBox: {
    gap: 20,
  },
  submit: {
    height: 50,
    backgroundColor: "#C6FFC9",
    borderRadius: 16,
    borderBlockColor: "black",
    borderWidth: 1,
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
  },
  submitText: {
    fontWeight: "600",
    fontSize: 18,
  },
  radioButtonBox: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-evenly",
  },
  radioButton: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
  },
  dropdownBox: {
    borderColor: "black",
    borderWidth: 1,
    borderRadius: 16,
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
  },
  dropdownText: {
    padding: 12,
    fontSize: 16,
  },
  dropdown: {
    padding: 12,
    width: "70%",
  },
});

export default NewIncomePage;
