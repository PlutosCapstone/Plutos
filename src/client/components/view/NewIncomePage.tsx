import React, { useState } from "react";
import { View, Text, StyleSheet, Pressable, TextInput } from "react-native";
import { RadioButton } from "react-native-paper";
import BackArrow from "../../assets/icons/BackArrow";

import { Dropdown } from "react-native-element-dropdown";
import { NavigationProps } from "../../types";
import IncomeService from "../../services/incomeService";
import { useUser } from "../../contexts/UserContext";

const dropdownData = [
  { label: "Daily", value: 1 },
  { label: "Weekly", value: 2 },
  { label: "Bi-Weekly", value: 3 },
  { label: "Monthly", value: 4 },
  { label: "Quarterly", value: 5 },
  { label: "Bi-annually", value: 6 },
  { label: "Annually", value: 7 },
];

interface NewIncomePageProps {
  navigation: NavigationProps;
}

type DropdownItem = {
  label: string;
  value: number;
};

const NewIncomePage = ({ navigation }: NewIncomePageProps) => {
  const [title, setTitle] = useState("");
  const [amount, setAmount] = useState("");
  const [checked, setChecked] = React.useState<"once" | "periodically">("once");
  const [dropdownValue, setDropdownValue] = useState(1);
  const { user } = useUser();

  const dropdownChangeHandler = (item: DropdownItem) => {
    setDropdownValue(item.value);
  };

  const grabLabelFromDropdownValue = (num: number) => {
    const entry = dropdownData.find((entry) => entry.value === num);
    return entry ? entry.label.toLowerCase() : null;
  };

  const returnHandler = () => {
    navigation.goBack();
  };

  const validationCheck = () => {
    const validAmount = Number(amount) > 0;
    const validTitle = title.length > 0;
    return validAmount && validTitle;
  };

  const addIncomeHandler = async () => {
    // probably put like an html indicator or smth idk
    if (!validationCheck()) return;

    const incomeData = {
      title,
      amount: Number(amount),
      frequency:
        checked === "once" ? null : grabLabelFromDropdownValue(dropdownValue),
      recurring: checked === "once" ? false : true,
      email: user!.email,
    };

    try {
      await IncomeService.createIncome(incomeData);
      returnHandler();
    } catch (error) {
      console.error("Error updating income:", error);
    }
  };

  return (
    <View style={styles.background}>
      <View style={styles.header}>
        <Pressable style={styles.backArrow} onPress={returnHandler}>
          <BackArrow size={35} />
        </Pressable>
        <Text style={styles.headerTitle}>New Income</Text>
      </View>
      <View style={styles.fieldBox}>
        <TextInput
          style={styles.textInput}
          value={title}
          onChangeText={setTitle}
          placeholder="Title: Add a Title"
        />
        <TextInput
          style={styles.textInput}
          onChangeText={setAmount}
          value={amount}
          placeholder="Amount ($):"
          keyboardType="numeric"
        />
        <View style={styles.radioButtonBox}>
          <View style={styles.radioButton}>
            <RadioButton
              value="once"
              status={checked === "once" ? "checked" : "unchecked"}
              onPress={() => setChecked("once")}
            />
            <Text onPress={() => setChecked("once")}>Once</Text>
          </View>
          <View style={styles.radioButton}>
            <RadioButton
              value="periodically"
              status={checked === "periodically" ? "checked" : "unchecked"}
              onPress={() => setChecked("periodically")}
            />
            <Text onPress={() => setChecked("periodically")}>Periodically</Text>
          </View>
        </View>
        {checked == "periodically" && (
          <View style={styles.dropdownBox}>
            <Text style={styles.dropdownText}>How often?:</Text>
            <Dropdown
              data={dropdownData}
              labelField="label"
              valueField="value"
              style={styles.dropdown}
              search
              searchPlaceholder="Search..."
              value={dropdownValue}
              onChange={(item) => dropdownChangeHandler(item)}
            />
          </View>
        )}

        <Pressable style={styles.submit} onPress={addIncomeHandler}>
          <Text style={styles.submitText}>Add Income</Text>
        </Pressable>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  background: {
    height: "100%",
    backgroundColor: "white",
    padding: 24,
    rowGap: 20,
  },
  header: {
    display: "flex",
    flexDirection: "row",
    position: "relative",
    justifyContent: "center",
    alignItems: "center",
  },
  backArrow: {
    position: "absolute",
    left: 0,
  },
  headerTitle: {
    fontWeight: "600",
    fontSize: 32,
  },
  textInput: {
    height: 50,
    borderBlockColor: "black",
    borderWidth: 1,
    borderRadius: 16,
    padding: 12,
    fontSize: 16,
  },
  fieldBox: {
    gap: 20,
  },
  submit: {
    height: 50,
    backgroundColor: "#C6FFC9",
    borderRadius: 16,
    borderBlockColor: "black",
    borderWidth: 1,
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
  },
  submitText: {
    fontWeight: "600",
    fontSize: 18,
  },
  radioButtonBox: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-evenly",
  },
  radioButton: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
  },
  dropdownBox: {
    borderColor: "black",
    borderWidth: 1,
    borderRadius: 16,
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
  },
  dropdownText: {
    padding: 12,
    fontSize: 16,
  },
  dropdown: {
    padding: 12,
    width: "70%",
  },
});

export default NewIncomePage;
