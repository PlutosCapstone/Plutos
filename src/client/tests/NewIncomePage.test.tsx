import { render, fireEvent } from "@testing-library/react-native";
import NewIncomePage from "../components/view/NewIncomePage";

describe("NewIncomePage component", () => {
  it("should match snapshot for new income page", () => {
    /* mock the navigation and route params */
    const navigation = jest.fn();
    const route = jest.fn();

    const tree = render(
      // @ts-ignore
      <NewIncomePage navigation={navigation} route={route} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should display all default visible UI elements", () => {
    /* mock the navigation and route params */
    const navigation = jest.fn();
    const route = jest.fn();

    const page = render(
      // @ts-ignore
      <NewIncomePage navigation={navigation} route={route} />,
    );

    /* technically should replace getByPlaceholderText with getByTestId but then
	we will need to add a test-id attribute to each component we're testing
	and thats a lotta work D: */
    const { getByText, getByPlaceholderText } = page;

    const header = getByText("New Income");
    const addIncome = getByText("Add Income");
    const title = getByPlaceholderText("Title: Add a Title");
    const amount = getByPlaceholderText("Amount ($):");
    const onceButton = getByText("Once");
    const periodicallyButton = getByText("Periodically");

    expect(header).toBeTruthy();
    expect(addIncome).toBeTruthy();
    expect(title).toBeTruthy();
    expect(amount).toBeTruthy();
    expect(onceButton).toBeTruthy();
    expect(periodicallyButton).toBeTruthy();
  });

  it("should update text fields when user inputs title/amount", () => {
    /* mock the navigation and route params */
    const navigation = jest.fn();
    const route = jest.fn();

    const page = render(
      // @ts-ignore
      <NewIncomePage navigation={navigation} route={route} />,
    );

    const { getByPlaceholderText } = page;

    const title = getByPlaceholderText("Title: Add a Title");
    fireEvent.changeText(title, "Gambling");
    expect(title.props.value).toBe("Gambling");
    expect(typeof title.props.value).toBe("string");

    const amount = getByPlaceholderText("Amount ($):");
    fireEvent.changeText(amount, 100);
    expect(amount.props.value).toBe(100);
    expect(typeof amount.props.value).toBe("number");
  });

  it("should show no dropdown options when 'once' is pressed", () => {
    /* mock the navigation and route params */
    const navigation = jest.fn();
    const route = jest.fn();

    const page = render(
      // @ts-ignore
      <NewIncomePage navigation={navigation} route={route} />,
    );

    const { getByTestId, queryByText } = page;

    const onceButton = getByTestId("once");
    const periodicallyButton = getByTestId("periodically");
    expect(onceButton).toBeTruthy();
    expect(periodicallyButton).toBeTruthy();

    fireEvent.press(onceButton);

    expect(onceButton.props.accessibilityState.checked).toBeTruthy();
    expect(periodicallyButton.props.accessibilityState.checked).toBeFalsy();

    const howOftenText = queryByText("How often?:");
    expect(howOftenText).toBeFalsy();
  });

  it("should show dropdown options when 'periodically' is pressed", () => {
    /* mock the navigation and route params */
    const navigation = jest.fn();
    const route = jest.fn();

    const page = render(
      // @ts-ignore
      <NewIncomePage navigation={navigation} route={route} />,
    );

    const { getByTestId, queryByText } = page;

    const onceButton = getByTestId("once");
    const periodicallyButton = getByTestId("periodically");
    expect(onceButton).toBeTruthy();
    expect(periodicallyButton).toBeTruthy();

    expect(onceButton.props.accessibilityState.checked).toBeTruthy();
    expect(periodicallyButton.props.accessibilityState.checked).toBeFalsy();

    fireEvent.press(periodicallyButton);

    expect(onceButton.props.accessibilityState.checked).toBeFalsy();
    expect(periodicallyButton.props.accessibilityState.checked).toBeTruthy();

    const howOftenText = queryByText("How often?:");
    expect(howOftenText).toBeTruthy();

    const occurrence = [
      "Daily",
      "Weekly",
      "Bi-Weekly",
      "Monthly",
      "Quarterly",
      "Bi-annually",
      "Annually",
    ];

    /* Default state when we choose periodically is set to 'Daily' by default */
    occurrence.forEach((val) => {
      if (val === "Daily") return expect(queryByText(val)).toBeTruthy();
      expect(queryByText(val)).toBeFalsy();
    });
  });
});
