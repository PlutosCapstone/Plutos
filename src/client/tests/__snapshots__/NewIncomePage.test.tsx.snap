// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewIncomeModal component should match snapshot for new income page 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#FFF",
          "borderRadius": 16,
          "elevation": 3,
          "padding": 24,
          "paddingBottom": 30,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 4,
          "width": "90%",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginBottom": 20,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "left": 0,
              "position": "absolute",
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={30}
            bbWidth={30}
            color="black"
            focusable={false}
            height={30}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {},
                {
                  "flex": 0,
                  "height": 30,
                  "width": 30,
                },
              ]
            }
            tintColor="black"
            vbHeight={512}
            vbWidth={512}
            width={30}
            xmlns="http://www.w3.org/2000/svg"
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            >
              <RNSVGPath
                d="M420.48 121.813L390.187 91.52L256 225.92L121.813 91.52L91.52 121.813L225.92 256L91.52 390.187l30.293 30.293L256 286.08l134.187 134.4l30.293-30.293L286.08 256z"
                fill={
                  {
                    "type": 2,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <Text
          style={
            {
              "fontSize": 24,
              "fontWeight": "600",
            }
          }
        >
          New Income
        </Text>
      </View>
      <View
        style={
          {
            "gap": 20,
          }
        }
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Title: Add a Title"
          style={
            {
              "borderColor": "black",
              "borderRadius": 16,
              "borderWidth": 1,
              "fontSize": 16,
              "height": 50,
              "padding": 12,
            }
          }
          value=""
        />
        <TextInput
          keyboardType="numeric"
          onChangeText={[Function]}
          placeholder="Amount ($):"
          style={
            {
              "borderColor": "black",
              "borderRadius": 16,
              "borderWidth": 1,
              "fontSize": 16,
              "height": 50,
              "padding": 12,
            }
          }
          value=""
        />
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-evenly",
            }
          }
        >
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              accessibilityLabel="Once"
              accessibilityRole="radio"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": true,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  false,
                  undefined,
                ]
              }
              testID="once"
            >
              <View
                pointerEvents="none"
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "paddingHorizontal": 16,
                      "paddingVertical": 8,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  accessibilityLiveRegion="polite"
                  accessibilityRole="radio"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": true,
                      "disabled": false,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      {
                        "borderRadius": 18,
                        "padding": 6,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "opacity": 1,
                      }
                    }
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      selectable={false}
                      style={
                        [
                          {
                            "color": "rgba(103, 80, 164, 1)",
                            "fontSize": 24,
                          },
                          [
                            {
                              "lineHeight": 24,
                              "transform": [
                                {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            {
                              "backgroundColor": "transparent",
                            },
                          ],
                          {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      󰄬
                    </Text>
                  </View>
                </View>
                <Text
                  style={
                    [
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": "400",
                          "letterSpacing": 0.15,
                          "lineHeight": 24,
                        },
                        [
                          {
                            "flexGrow": 1,
                            "flexShrink": 1,
                          },
                          false,
                          {
                            "color": "rgba(28, 27, 31, 1)",
                            "textAlign": "right",
                          },
                          undefined,
                        ],
                      ],
                    ]
                  }
                >
                  Once
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              accessibilityLabel="Periodically"
              accessibilityRole="radio"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": false,
                  "disabled": false,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  false,
                  undefined,
                ]
              }
              testID="periodically"
            >
              <View
                pointerEvents="none"
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "paddingHorizontal": 16,
                      "paddingVertical": 8,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  accessibilityLiveRegion="polite"
                  accessibilityRole="radio"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": false,
                      "disabled": false,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      {
                        "borderRadius": 18,
                        "padding": 6,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "opacity": 0,
                      }
                    }
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      selectable={false}
                      style={
                        [
                          {
                            "color": "rgba(103, 80, 164, 1)",
                            "fontSize": 24,
                          },
                          [
                            {
                              "lineHeight": 24,
                              "transform": [
                                {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            {
                              "backgroundColor": "transparent",
                            },
                          ],
                          {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      󰄬
                    </Text>
                  </View>
                </View>
                <Text
                  style={
                    [
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": "400",
                          "letterSpacing": 0.15,
                          "lineHeight": 24,
                        },
                        [
                          {
                            "flexGrow": 1,
                            "flexShrink": 1,
                          },
                          false,
                          {
                            "color": "rgba(28, 27, 31, 1)",
                            "textAlign": "right",
                          },
                          undefined,
                        ],
                      ],
                    ]
                  }
                >
                  Periodically
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#C6FFC9",
              "borderColor": "black",
              "borderRadius": 16,
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 18,
                "fontWeight": "600",
              }
            }
          >
            Add Income
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
